### PR TITLE
@W-8888184 - Refactor public commands.

### PR DESCRIPTION
**Pull request by sfdctaka**
Incorporating the refactor done in the `core` lib into public commands.

### CODE REVIEW

File: src/cli/commands/force/lightning/local/device/create.ts [hunk 1]
```diff
@@ -165,6 +163,20 @@ export class Create extends Setup {
...
+            const commandDict: CommandRequirements = {};
+            commandDict.setup = getPlatformSetupRequirements(
+                this.logger,
+                this.flags.platform,
+                this.flags.apilevel
+            );
+            this._requirements = commandDict;
+        }
+        return this._requirements;
+    }
```
Comment (sfdctaka) [inline src/cli/commands/force/lightning/local/device/create.ts:178]: For now each command has its own private `_requirements` as the base class `Setup`'s property is private and cannot be accessed.

File: src/cli/commands/force/lightning/setupRequirementsUtil.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+export function getPlatformSetupRequirements(
+    logger: Logger,
+    platform: string,
+    apiLevel?: string
+): RequirementList {
+    return CommandLineUtils.platformFlagIsAndroid(platform)
+        ? new AndroidEnvironmentRequirements(logger, apiLevel)
+        : new IOSEnvironmentRequirements(logger);
+}
```
Comment (sfdctaka) [inline src/cli/commands/force/lightning/setupRequirementsUtil.ts:15]: I made a helper function which may go away after the next round of refactoring in `core`. This code might stay if we don't use `Setup` or another base command to inherit from(which I'm leaning more into). For now though I need this method because all commands need the same set of requirements as `Setup`.

File: src/cli/commands/force/lightning/local/device/create.ts [hunk 3]
```diff
@@ -165,6 +163,20 @@ export class Create extends Setup {
             });
     }
 
+    private _requirements: CommandRequirements = {};
+    public get commandRequirements(): CommandRequirements {
+        if (Object.keys(this._requirements).length === 0) {
+            const commandDict: CommandRequirements = {};
```
Comment (maliroteh-sf) [inline src/cli/commands/force/lightning/local/device/create.ts:169]: nit: rename `commandDict` to `requirements` since that's more representative of what it is (in here and other files)

File: src/cli/commands/force/lightning/local/device/list.ts [hunk 4]
```diff
@@ -48,16 +53,17 @@ export class List extends Setup {
         PerformanceMarkers.FETCH_DEVICES_MARKER_KEY
     )!;
 
-    public async run(direct: boolean = false): Promise<any> {
-        if (direct) {
-            await this.init(); // ensure init first
-        }
+    public async run(): Promise<any> {
```
Comment (maliroteh-sf) [inline src/cli/commands/force/lightning/local/device/list.ts:56]: The `List` command should not change for now and should not run setup requirements at all. We use this command in our VSCode extension and we don't have a way there to tell the command to skip its requirement checks, which is why we decided to have this command to not do checks at all. You should revert the changes to this file (and its unit tests) for now.

File: src/cli/commands/force/lightning/local/device/create.ts [hunk 5]
```diff
@@ -199,79 +211,3 @@ export class Create extends Setup {
         );
     }
 }
-
```
Comment (maliroteh-sf) [inline src/cli/commands/force/lightning/local/device/create.ts:202]: I'm confused... why are these extra requirement checks being removed? Similar to the `Preview` command, the `Create` command needs the base requirement checks + these extra checks.

**Review by maliroteh-sf — CHANGES_REQUESTED** [review]

File: src/cli/commands/force/lightning/local/device/create.ts (hunk 5 above)
Comment (sfdctaka) [reply src/cli/commands/force/lightning/local/device/create.ts:202]: The classes below aren't used anywhere. I can put it back. Can you tell me though where they are referenced?

Comment (maliroteh-sf) [reply src/cli/commands/force/lightning/local/device/create.ts:202]: They were used in lines 73-77 as `additionalRequirements`, which you removed:

```
        const extraReqs: Requirement[] = [
            new DeviceNameAvailableRequirement(this, this.logger),
            new ValidDeviceTypeRequirement(this, this.logger)
        ];
        this.addAdditionalRequirements(extraReqs);
```

Comment (sfdctaka) [reply src/cli/commands/force/lightning/local/device/create.ts:202]: 😬 
Thanks!

File: src/cli/commands/force/lightning/local/device/list.ts (hunk 4 above)
Comment (sfdctaka) [reply src/cli/commands/force/lightning/local/device/list.ts:56]: I didn't do a full revert but removed running requirement processor and updated the test accordingly.

File: src/cli/commands/force/lightning/local/device/create.ts [hunk 6]
```diff
@@ -165,6 +164,27 @@ export class Create extends Setup {
...
+                this.logger,
+                this.flags.platform,
+                this.flags.apilevel
+            );
+            requirements.create = {
+                requirements: [
+                    new DeviceNameAvailableRequirement(this, this.logger),
+                    new ValidDeviceTypeRequirement(this, this.logger)
+                ],
+                enabled: true
```
[the remark this answers is not in the document]
Comment (sfdctaka) [reply src/cli/commands/force/lightning/local/device/create.ts:181]: Preview is a bit different because we need to wait until the runtime to get the flag value. That said, the next round of refactor work shall address this issue and provide more consistent approach. If you want I can still do the same as preview but the primary goal for this PR was to get it functional with the latest `core`. Let me know!

**Review by maliroteh-sf — APPROVED** [review]
LGTM 👍
